src/transforms/filter.jl

```diff
@@ -6,6 +6,24 @@
     Filter(function)
 
 Filters the table returning only the rows where the `function` returns true.
+
+# Examples
+```julia
+T = Filter(row -> sum(row) > 10)
+T = Filter(row -> row.a == true && row.b < 30)
+```
+
+Note that schema of the new table is the same as the original table:
+```julia
+julia> table = (a = [1, missing, 2, 3], b = [missing, 1, 2, 3]);
+
+julia> T = Filter(row -> all(!ismissing, row));
+
+julia> newtable, cache = apply(T, table);
+
+julia> newtable
+(a = Union{Missing, Int64}[2, 3], b = Union{Missing, Int64}[2, 3])
+```
```

Comment:
```suggestion
```

This note should be written as a small comment in a ## Notes section. I think we have examples of notes in this repo already. If not, take a look at the GeoStats.jl or Meshes.jl projects where we have a bunch of notes in docstrings.

Notes shouldn't be the first thing a user reads, these are very technical details that should only come up at the end for advanced users trying to grasp nuances.